spec/models/validations/sales/setup_validations_spec.rb

```diff
@@ -26,29 +17,41 @@
         end
       end
 
-      context "when saledate is in the 22/23 collection year" do
+      context "when saledate is in the open collection year" do
         let(:record) { build(:sales_log, saledate: Time.zone.local(2023, 1, 1)) }
 
+        before do
+          allow(Time).to receive(:now).and_return(Time.zone.local(2023, 1, 10))
+        end
+
         it "does not add an error" do
           setup_validator.validate_saledate_collection_year(record)
 
           expect(record.errors).to be_empty
         end
       end
 
-      context "when saledate is before the 22/23 collection year" do
+      context "when saledate is before the open collection year" do
         let(:record) { build(:sales_log, saledate: Time.zone.local(2020, 1, 1)) }
 
+        before do
+          allow(Time).to receive(:now).and_return(Time.zone.local(2023, 1, 10))
+        end
```

Comment:
Similarly to above, could these be relative to real now rather than needing to fix dates?

I think we could use a regex to avoid needing to specify specific years in the error message assertion.

Reply:
I feel like this one is a bit trickier, cause we're testing the error message content `Enter a date within the 22/23 collection year`, I think I'd be inclined to leave this, unless we construct the error message on current open period dates, but then I feel like we're not testing that the error message gets constructed correctly cause we'd be mirroring the code that we're testing (?)

Reply:
In one of the tests on my new PR I have something like this and I decided that it was sufficient to test that an error was being added and it had the right format using regex.
```
expect(record.errors[:saledate]).to include(/Enter a date within the \d{2}\/\d{2} collection year, which is between 1st April \d{4} and 31st March \d{4}/)
```

Reply:
I'll rebase when merged!